src/System.CommandLine/Invocation/TypoCorrection.cs

```diff
@@ -23,9 +23,10 @@ public TypoCorrection(int maxLevenshteinDistance)
 
         public void ProvideSuggestions(ParseResult result, IConsole console)
         {
-            for (var i = 0; i < result.UnmatchedTokens.Count; i++)
+            var unmatchedTokens = result.UnmatchedTokens;
+            for (var i = 0; i < unmatchedTokens.Count; i++)
             {
-                var token = result.UnmatchedTokens[i];
+                var token = unmatchedTokens[i];
                 var suggestions = GetPossibleTokens(result.CommandResult.Command, token).ToList();
                 if (suggestions.Count > 0)
                 {
```

Comment:
This is one less allocation, right?
```suggestion
                var suggestions = GetPossibleTokens(result.CommandResult.Command, token);
                if (suggestions.Any())
                {
```

Reply:
Actually there was a simpler solution: https://github.com/dotnet/command-line-api/pull/2001/commits/96c8e77a6334b2d7ba93b4d71e37f26f33f9ae24